### PR TITLE
API keys: Respect CACHE_WRITE capability in remote cache

### PR DESCRIPTION
**Pull request by bduffany**
Tested with these commands:

```bash
cd buildbuddy-io/buildbuddy

# Make the API key read-only
sqlite3 /tmp/buildbuddy-enterprise.db \
  'UPDATE APIKeys WHERE value = "OBZUMNQtiQ550j2tfHNX" SET capabilites = 0;'

# Build once with remote cache config.
# Build succeeds with no errors in console.
# "Cache" tab in BB UI shows 0 bytes up/down.
# Local cache directory for the authenticated group is empty.
bazel clean && bazel build //server \
  --remote_header=x-buildbuddy-api-key=OBZUMNQtiQ550j2tfHNX \
  --remote_cache=grpc://localhost:1985 \
  --bes_backend=grpc://localhost:1985 \
  --bes_results_url=http://localhost:8080/invocation/ \
  --remote_upload_local_results

# Run the same command again.
# Build runs slowly, as expected. Local cache dir is stil empty.

# Now run the same command with a different API key with CACHE_WRITE caps.
# Results are uploaded to cache, as expected.

# Now run the command with the read-only key again.
# Build runs very quickly (in under 2s) since the cached results are read.
```

---

**Version bump**: Minor 

**Related issues**: https://github.com/buildbuddy-io/buildbuddy-internal/issues/213

### CODE REVIEW

**Comment by bduffany** [comment]
Note to self: Change base branch to `master` before merging

**Review by tylerwilliams — CHANGES_REQUESTED** [review]
Cache changes mostly LG

File: server/util/capabilities/capabilities.go [hunk 1]
```diff
@@ -21,3 +36,22 @@ func ToInt(caps []akpb.ApiKey_Capability) int32 {
 	}
 	return m
 }
+
+func IsGranted(ctx context.Context, env environment.Env, cap akpb.ApiKey_Capability) (bool, error) {
```
[the remark this answers is not in the document]
Comment (bduffany) [reply server/util/capabilities/capabilities.go:47]: Done!

File: server/util/capabilities/BUILD [hunk 2]
```diff
@@ -7,15 +7,34 @@ go_library(
...
         "//proto:api_key_go_proto",
+        "//server/interfaces:go_default_library",
+        "//server/testutil/auth:go_default_library",
+        "//server/testutil/environment:go_default_library",
+        "//server/util/testing/flags:go_default_library",
         "@com_github_stretchr_testify//assert:go_default_library",
+        "@org_golang_google_grpc//codes:go_default_library",
+        "@org_golang_google_grpc//status:go_default_library",
     ],
 )
```
Comment (tylerwilliams) [inline server/util/capabilities/BUILD:40]: is gazelle going to overwrite this?

File: server/testutil/auth/auth.go [hunk 3]
```diff
@@ -7,11 +7,12 @@ import (
 
 	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
 	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/util/capabilities"
 	"github.com/buildbuddy-io/buildbuddy/server/util/status"
 
 	"google.golang.org/grpc/metadata"
 
-	akpb "github.com/buildbuddy-io/buildbuddy/proto/api_key"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/auth"
```
Comment (tylerwilliams) [inline server/testutil/auth/auth.go:15]: it strikes me as a little error-prone to import this into a package with the same name. Can we call it realauth or something?

**Review by tylerwilliams — APPROVED** [review]
LGTM (two small nits)

File: server/util/capabilities/BUILD (hunk 2 above)
Comment (bduffany) [reply server/util/capabilities/BUILD:40]: good catch, added `# keep` comments, and gazelle doesn't overwrite it anymore.

File: server/testutil/auth/auth.go (hunk 3 above)
Comment (bduffany) [reply server/testutil/auth/auth.go:15]: Done

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**